scene/resources/dynamic_font.cpp

```diff
@@ -303,14 +328,182 @@ void DynamicFontAtSize::set_texture_flags(uint32_t p_flags) {
 	}
 }
 
+bool DynamicFontAtSize::shape_run(int p_index, int32_t p_pre_shape_clusters, Vector<Ref<ClusterInfo> > &p_visual_order, Vector<Ref<ClusterInfo> > &p_logical_order, const Vector<UChar> &p_utf16_text, int32_t p_start, int32_t p_length, TextDirection p_direction, const Vector<hb_feature_t> &p_features, hb_script_t p_script, hb_language_t p_lang) const {
+
+	if (p_length <= 0)
+		return false;
+
+	hb_buffer_t *buffer = hb_buffer_create();
+	hb_buffer_set_direction(buffer, (p_direction == DIR_LTR) ? HB_DIRECTION_LTR : HB_DIRECTION_RTL);
+
+	if (p_script) hb_buffer_set_script(buffer, p_script);
+	if (p_lang) hb_buffer_set_language(buffer, p_lang);
```

Comment:
Not setting a language will cause HarfBuzz to use the current locale, which is OK for a UI toolkit (just making sure you are aware of this).